frappe/utils/response.py

```diff
@@ -199,7 +199,7 @@ def _make_logs_v1():
 			[orjson.dumps(d).decode() for d in frappe.local.message_log]
 		).decode()
 
-	if frappe.debug_log:
+	if frappe.debug_log and is_traceback_allowed():
 		response["_debug_messages"] = orjson.dumps(frappe.local.debug_log).decode()
```

Comment:
I think there it's okay to send the debug messages (it is used by the System Console for print logs for example)

The security issue might be deeper:

<img width="1221" height="551" alt="image" src="https://github.com/user-attachments/assets/7d1dce98-5de6-42f2-8cce-1827792e39e8" />